include/vapor/DC.h

```diff
@@ -641,6 +641,12 @@ class VDF_API DC : public Wasp::MyBase {
         //
         size_t GetTopologyDim() const;
 
+        //! Generate a mesh name from a list of strings. The 
+        //! generated name is simply the join of the elements of \p s
+        //! using 'x' as a delimiter
+        //
+        static string MakeMeshName(std::vector<string> s);
```

Comment:
Why does this function have to be static?

Reply:
So it can be used without instantiating a class as is done in DCCF.cpp

Reply:
This this the use case that is required occasionally: a helper function that takes input and produces an output, and the operation is not necessarily associated with a particular class. 

There are at least 2 ways to achieve the goal: 1) make it a function under a particular namespace. E.g.: many functions under the `std::` namespace in C++. 2) declare the function static and insert it into a particular class.

Way 2) is a little messy, because the static functions will out-live the class object itself, which is really against the core ideas of object oriented programming. Way 1) is adapted by the standard C++ to meet this particular requirement, so I suspect it's got other advantages. 

I suggest adopting Way 1) here, but it's ultimately the decision of the DC author.

Reply:
This method has no purpose that isn't associated with the DC class (i.e. it will NOT out-live the class. That is why it was put into the same namespace. Putting it in some other made up namespace would only cause confusion. I'm inclined to leave it where it is.

Reply:
From a design perspective, if all purposes of a function is associated with a particular class, then wouldn't it make sense to have this function be a member function of that class?

Reply:
It is a member function of a class. It's a static member function. One of the intended use cases for static member functions is to allow their use without instantiating the class.